### DISCUSSION TITLE
beets config path

### DISCUSSION THREAD

**Discussion by Kolbasz12** [RandomNinjaAtk/docker-lidarr-extended#107] [General]
beet config -p reveals:
/root/.config/beets/config.yaml

This path will overwrite with container restart, Can this be changed to point to /config?  Want to enable custom config settings for beets.

**Comment by RandomNinjaAtk** [comment]
You should put your file where you want, and use a method similar to the below reference to run it:
https://github.com/RandomNinjaAtk/docker-lidarr-extended/blob/03261633a6bef5d56b25b147bfe0c1475ad426f4/root/scripts/audio.sh#L1740

**Reply by Kolbasz12** [reply]
so that is the beet import call that is now disabled by default correct?  Is there a way to enable that if we want to use it still and have that change maintained through updates?

If not, I will look to call that same command using a .sh via the connect options.

Can you clarify where $1 is defined and also what &>/dev/null  means in that line, beyond, "This is just an abbreviation for >/dev/null 2>&1. It redirects file descriptor 2 (STDERR) and descriptor 1 (STDOUT) to /dev/null."

**Reply by RandomNinjaAtk** [reply]
The command I referenced for you was an example of how the "beet" command was used in the script, you'll need to write you own command to use beets, its best to review beets documentation. The line demonstrates using global flags for specifying config file location, library location and etc.

All of the beets specific scripting code was removed, so there isn't a way to re-enable it because it doesn't exist.

Beets documentation: https://beets.readthedocs.io/en/stable/reference/cli.html

**Reply by Kolbasz12** [reply]
Actually.  Just realized an issue with the above suggestion as all plugins are missing.  Beets will import the plugins on start, but calling the remote config, plugins are missing and beets errors out.

Is there a way to customize this or are we stuck?  thanks.

` beet -c /config/beets/config.yaml -l /scripts/library.blb import /music/Metallica/
** error loading plugin lastgenre:
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/beets/plugins.py", line 268, in load_plugins
    namespace = __import__(modname, None, None)
  File "/usr/lib/python3.9/site-packages/beetsplug/lastgenre/__init__.py", line 24, in <module>
    import pylast
ModuleNotFoundError: No module named 'pylast'

** error loading plugin web:
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/beets/plugins.py", line 268, in load_plugins
    namespace = __import__(modname, None, None)
  File "/usr/lib/python3.9/site-packages/beetsplug/web/__init__.py", line 21, in <module>
    import flask
ModuleNotFoundError: No module named 'flask'

** error loading plugin discogs:
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/beets/plugins.py", line 268, in load_plugins
    namespace = __import__(modname, None, None)
  File "/usr/lib/python3.9/site-packages/beetsplug/discogs.py", line 24, in <module>
    from discogs_client import Release, Master, Client
ModuleNotFoundError: No module named 'discogs_client'

** error loading plugin beatport:
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/beets/plugins.py", line 268, in load_plugins
    namespace = __import__(modname, None, None)
  File "/usr/lib/python3.9/site-packages/beetsplug/beatport.py", line 22, in <module>
    from requests_oauthlib import OAuth1Session
ModuleNotFoundError: No module named 'requests_oauthlib'

error: replaygain initialization failed: no replaygain command found: install mp3gain or aacgain`

**Reply by RandomNinjaAtk** [reply]
Looks like its failing because the plugin dependencies are not installed, you'll need to include installing the plugin dependencies you plan to use in your script.

The problem is not the config file location.

Edit: You'll want to review each plugins documentation for requirements and etc...

**Reply by Kolbasz12** [reply]
hmm, got it.  I will play with it.  I am trying to combine beets and lidarr into one.  Thats the goal anyway.

Curious, I use a lot of linuxserver containers and they support a custom config, where you can run commands on container start.  Does this container support something similar?  If not, thats OK, just want to make sure I am not chasing something that is not possible.

https://www.linuxserver.io/blog/2019-09-14-customizing-our-containers

**Reply by RandomNinjaAtk** [reply]
This container base is the same as theirs, so yes it does

**Reply by Kolbasz12** [reply]
OK, thanks, I will give it a shot.  I managed to manually install most plugins, the only one still failing is replaygain.

either: 
error: replaygain initialization failed: Failed to load GStreamer: python-gi not found
https://github.com/linuxserver/docker-beets/issues/20

or:
error: replaygain initialization failed: no replaygain command found: install mp3gain or aacgain

In the meantime, I can remove replain gain from the config.

**Comment by RandomNinjaAtk** [comment]
Just FYI, I have re-integrated beets as an optional process. Still similar to its original integration.

**Comment by Kolbasz12** [comment]
Does it now auto call or is it still via a connection?  Since it was disabled, what was the original implementation?

**Reply by RandomNinjaAtk** [reply]
https://github.com/RandomNinjaAtk/docker-lidarr-extended#parameters
